### ISSUE TITLE
Uploading images shows error — Your message was not bridged: failed to reupload media

### ISSUE THREAD

**Issue by evoL** [mautrix/signal#564]
Since yesterday I'm unable to send attachments over the Signal bridge. Uploading an image in a 1:1 chat shows me an `Your message was not bridged: failed to reupload media` error.

<img width="1228" alt="image" src="https://github.com/user-attachments/assets/7602cdce-be11-45ba-a594-b4d6a6b04de5">

This is happening on mautrix-signal v0.7.1 and v0.7.2. From the logs, it looks like we're hitting a HTTP 404 on `https://chat.signal.org/v3/attachments/form/upload`.

Logs:

```
signal-1  | 2024-10-18T08:43:52.122Z DBG Event decrypted successfully action="decrypt megolm event" decrypted_event_type="m.room.message (message)" event_id=$ACRnRhgGz7_eUzsjhYxYMaodgc7BIqewujIe2MdEG_U message_index=1 sender=@evol:evolved.systems sender_key=mrjDhSHEzr+tUYmYm65Eqo1xgHKuMb7gztufsaNDWAM session_id=TV8bQSbwzBjyIF0VwXH/RY2rt+XgalH/+fDgGaze+WA
signal-1  | 2024-10-18T08:43:52.132Z DBG Request completed action="handle matrix event" duration=9.806392 event_id=$ACRnRhgGz7_eUzsjhYxYMaodgc7BIqewujIe2MdEG_U event_loop_index=4 event_type=m.room.message login_id=5aa3aa5a-a594-4aa7-83b0-568a3e7b35fa method=GET portal_id=5aa3aa5a-a594-4aa7-83b0-568a3e7b35fa portal_mxid=!SjHhtSBSiLGSEnolGE:evolved.systems req_id=25 response_length=-1 response_mime=application/octet-stream sender=@evol:evolved.systems status_code=200 url=http://matrix:8008/_matrix/client/v1/media/download/evolved.systems/WXysRwqxAlNsXrtZpmoNgjiD?user_id=%40signalbot%3Aevolved.systems
signal-1  | 2024-10-18T08:43:52.451Z DBG received HTTP response action="send HTTP request" event_id=$ACRnRhgGz7_eUzsjhYxYMaodgc7BIqewujIe2MdEG_U event_loop_index=4 event_type=m.room.message login_id=5aa3aa5a-a594-4aa7-83b0-568a3e7b35fa method=GET portal_id=5aa3aa5a-a594-4aa7-83b0-568a3e7b35fa portal_mxid=!SjHhtSBSiLGSEnolGE:evolved.systems request_number=9 sender=@evol:evolved.systems status_code=404 url=https://chat.signal.org/v3/attachments/form/upload
signal-1  | 2024-10-18T08:43:52.451Z DBG unexpected status code action="handle matrix event" body="{\"code\":404,\"message\":\"HTTP 404 Not Found\"}" event_id=$ACRnRhgGz7_eUzsjhYxYMaodgc7BIqewujIe2MdEG_U event_loop_index=4 event_type=m.room.message login_id=5aa3aa5a-a594-4aa7-83b0-568a3e7b35fa portal_id=5aa3aa5a-a594-4aa7-83b0-568a3e7b35fa portal_mxid=!SjHhtSBSiLGSEnolGE:evolved.systems sender=@evol:evolved.systems status_code=404
signal-1  | 2024-10-18T08:43:52.451Z ERR Error decoding response body fetching upload attributes error="Unexpected status code: 404 404 Not Found" action="handle matrix event" event_id=$ACRnRhgGz7_eUzsjhYxYMaodgc7BIqewujIe2MdEG_U event_loop_index=4 event_type=m.room.message func="upload attachment" login_id=5aa3aa5a-a594-4aa7-83b0-568a3e7b35fa portal_id=5aa3aa5a-a594-4aa7-83b0-568a3e7b35fa portal_mxid=!SjHhtSBSiLGSEnolGE:evolved.systems sender=@evol:evolved.systems
signal-1  | 2024-10-18T08:43:52.451Z ERR Failed to upload file error="Unexpected status code: 404 404 Not Found" action="handle matrix event" event_id=$ACRnRhgGz7_eUzsjhYxYMaodgc7BIqewujIe2MdEG_U event_loop_index=4 event_type=m.room.message login_id=5aa3aa5a-a594-4aa7-83b0-568a3e7b35fa portal_id=5aa3aa5a-a594-4aa7-83b0-568a3e7b35fa portal_mxid=!SjHhtSBSiLGSEnolGE:evolved.systems sender=@evol:evolved.systems
signal-1  | 2024-10-18T08:43:52.451Z ERR Failed to handle Matrix message error="failed to convert attachment: failed to reupload media: Unexpected status code: 404 404 Not Found" action="handle matrix event" event_id=$ACRnRhgGz7_eUzsjhYxYMaodgc7BIqewujIe2MdEG_U event_loop_index=4 event_type=m.room.message login_id=5aa3aa5a-a594-4aa7-83b0-568a3e7b35fa portal_id=5aa3aa5a-a594-4aa7-83b0-568a3e7b35fa portal_mxid=!SjHhtSBSiLGSEnolGE:evolved.systems sender=@evol:evolved.systems
```

**Comment by evoL** [comment]
I should add that sending pictures over the Signal app works fine (as expected). The messages from Signal are also bridged over to Matrix without issues.

**Comment by evoL** [comment]
I guess that this is already mitigated by https://mau.dev/mautrix/signal/-/commit/b09995a892c9930628e1669532d9c1283a4938c8.

**Comment by EchedelleLR** [comment]
@evoL I guess we need to wait for next release?

**Comment by mikemrm** [comment]
I pulled latest docker main build, worked perfectly. `dock.mau.dev/mautrix/signal:ad41c86f116f484a57b7df07bc082f33fc12d908-amd64`

**Comment by EchedelleLR** [comment]
@derritter88 you have to wait for next release, by now is only in the main branch, not tagged.

**Comment by derritter88** [comment]
@EchedelleLR Yeah thought so.
Would a downgrade be supported as on the previous version it worked well

**Comment by tulir** [comment]
You don't have to wait for the next release, you can use the fixed version at any time.

Downgrading won't help.

**Comment by tulir** [comment]
Anything except option 2 in https://docs.mau.fi/bridges/go/setup.html?bridge=signal#step-1-installation (or if using docker, the `latest` tag)

**Comment by anytimesoon** [comment]
I'm still getting this error on v0.7.2

**Comment by slagiewka** [comment]
> I'm still getting this error on v0.7.2

v0.7.2 was released BEFORE the commit fixing the issue was made. You need to wait for 0.7.3 or 0.8.0 or use a build from master.

**Comment by anytimesoon** [comment]
Oh I see. I had misunderstood the thread. Thank you for clarifying

**Comment by tom-ch1** [comment]
confirmed fixed in Version 0.7.3
